src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/CodeActionTest`1.cs

```diff
@@ -93,6 +94,37 @@ protected static bool HasAnyChange(SolutionState oldState, SolutionState newStat
             }
         }
 
+        protected virtual ImmutableArray<CodeAction> FilterCodeActions(ImmutableArray<CodeAction> actions)
+        {
+            var builder = actions.ToBuilder();
+            while (true)
```

Comment:
Will this ever return if there are no nested code actions?

Reply:
Yes, it will return. `changesMade` is initialized to false, and the loop only continues when processing an action sets it to true.